### PR TITLE
Adding section about GeoJSON clustering

### LINKED ISSUE #505: Add mention/section about GeoJSON runtime styling clustering
[Because the cluster util plugin is being deprecated](https://github.com/mapbox/mapbox-plugins-android/issues/535#event-1642395945), would be good to eventually add some sort of mention/section about runtime styling clustering with GeoJSON to  https://www.mapbox.com/android-docs/map-sdk/overview/runtime-styling/

### PR DESCRIPTION
**Pull request by langsmith**
Resolves #505 by adding more written docs about using runtime/data-driven styling to do (GeoJSON) clustering. 

Related to the demo app's examples:

- https://github.com/mapbox/mapbox-android-demo/blob/master/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/GeoJsonClusteringActivity.java
- https://github.com/mapbox/mapbox-android-demo/blob/master/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/ImageClusteringActivity.java

screencapture-localhost-8080-android-docs-map-sdk-overview-clustering-2018-07-02-16_20_01

### CODE REVIEW

**Comment by langsmith** [comment]
cc @tobrun @LukasPaczos (feel free to give feedback, add writing, etc.)

File: src/pages/map-sdk/overview/clustering.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Often times, too much data is shown on a map at a single time. Markers overlap with each other. The map looks and feels cluttered. Users can't get a quick understanding of what the data is supposed to say. 
+
+Showing clustered data is entirely possible by using the runtime/data-driven styling capabilites of the Mapbox Maps SDK for Android.
+
+Tying the amount of data shown on the map to the map's camera zoom level, is a great way to provide users with a cleaner UI experience and less overwhelming location data experience. 
+	
+
+To accomplish this, you:
+	
+1. Will have a some sort of dataset, preferably GeoJSON.
```
Comment (tobrun) [inline src/pages/map-sdk/overview/clustering.md]: afaik only geojson is supported and a reference to GeoJsonSource would be nice

Comment (LukasPaczos) [reply src/pages/map-sdk/overview/clustering.md]: shouldn't `VectorSource` work as well @tobrun?

Comment (langsmith) [reply src/pages/map-sdk/overview/clustering.md]: >afaik only geojson is supported and a reference to GeoJsonSource would be nice

Ok, added

File: src/pages/map-sdk/overview/clustering.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+prependJs:
+  - import { AppropriateImage } from '../../../components/appropriate-image';
+  - "import { Floater } from '../../../components/floater';"
+---
+
+# Data clustering
+
+Often times, too much data is shown on a map at a single time. Markers overlap with each other. The map looks and feels cluttered. Users can't get a quick understanding of what the data is supposed to say. 
+
+Showing clustered data is entirely possible by using the runtime/data-driven styling capabilites of the Mapbox Maps SDK for Android.
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/map-sdk/overview/clustering.md]: removed

File: src/pages/map-sdk/overview/clustering.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+Tying the amount of data shown on the map to the map's camera zoom level, is a great way to provide users with a cleaner UI experience and less overwhelming location data experience.
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/map-sdk/overview/clustering.md]: Fixed

File: src/pages/map-sdk/overview/clustering.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+Often times, too much data is shown on a map at a single time. Markers overlap with each other. The map looks and feels cluttered. Users can't get a quick understanding of what the data is supposed to say. 
+
+Showing clustered data is entirely possible by using the runtime/data-driven styling capabilites of the Mapbox Maps SDK for Android.
+
+Tying the amount of data shown on the map to the map's camera zoom level, is a great way to provide users with a cleaner UI experience and less overwhelming location data experience. 
+	
+
+To accomplish this, you:
+	
+1. Will need a GeoJSON dataset and add it to the Mapbox map as a `GeoJsonSource`.
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/map-sdk/overview/clustering.md]: Fixed

[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/map-sdk/overview/clustering.md]: Fixed

File: src/pages/map-sdk/overview/clustering.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	            .withClusterMaxZoom(MAX_ZOOM)
+	            .withClusterRadius(DESIRED_CLUSTER_RADIUS)
+	        )
+	      );
+	    } catch (MalformedURLException malformedUrlException) {
+	      Log.e(TAG, "Check the URL " + malformedUrlException.getMessage());
+	    }
+	```
+	
+2. Create a `SymbolLayer` with icons that represent the individual data points for when there is no clustering happening. These icons will only be visible when the map's camera is close enough to the map (i.e. zoomed in). Remember, the higher the map zoom value, the more zoomed in the camera is. A zoom level of 12 is closer to the map than a zoom value of 4.
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/map-sdk/overview/clustering.md]: Fixed

[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/map-sdk/overview/clustering.md]: Removed

File: src/pages/map-sdk/overview/clustering.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+3. Create as many additional `SymbolLayer`s or `CircleLayer`s as you want for the various data ranges. You might have red circles represent data clusters that have 10-30 data points and then blue circles that have 50 or more data points. Data-driven styling and `Expression` filtering will determine which cluster layers are shown at what time.
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/map-sdk/overview/clustering.md]: Fixed

**Comment by langsmith** [comment]
Feedback has been addressed @colleenmcginnis

**Comment by colleenmcginnis** [comment]
@langsmith I made a few adjustments for the new docs site setup. The images you've included here are very small, and as a result are looking quite pixelated. Any chance you could get a higher res version of these screenshots?

**Comment by langsmith** [comment]
I'll send you the raw screenshot images @colleenmcginnis

**Comment by colleenmcginnis** [comment]
☝️ Updated circle layer image @langsmith

**Pull request merged**